src/smt/store/smt_store.rs

```diff
@@ -0,0 +1,172 @@
+//! Implement SMTStore trait
+
+use crate::smt::db::cota_db::CotaRocksDB;
+use crate::smt::db::schema::Col;
+use crate::smt::store::serde::leaf_key_to_vec;
+use crate::utils::error::Error;
+use crate::utils::helper::parse_vec_n;
+use cota_smt::smt::H256;
+use serde::{Deserialize, Serialize};
+use sparse_merkle_tree::{
+    error::Error as SMTError,
+    traits::Store,
+    tree::{BranchKey, BranchNode},
+};
+
+use super::serde::{branch_key_to_vec, branch_node_to_vec, slice_to_branch_node};
+
+#[derive(Clone, Eq, PartialEq, Serialize, Deserialize)]
+struct SMTLeaf {
+    key:   Vec<u8>,
+    value: Vec<u8>,
+}
+
+pub struct SMTStore<'a> {
+    lock_hash:  [u8; 32],
+    leaf_col:   Col,
+    branch_col: Col,
+    root_col:   Col,
+    leaves_col: Col,
+    store:      &'a CotaRocksDB,
+}
+
+impl<'a> SMTStore<'a> {
+    pub fn new(
+        lock_hash: [u8; 32],
+        leaf_col: Col,
+        branch_col: Col,
+        root_col: Col,
+        leaves_col: Col,
+        store: &'a CotaRocksDB,
+    ) -> Self {
+        SMTStore {
+            lock_hash,
+            leaf_col,
+            branch_col,
+            root_col,
+            leaves_col,
+            store,
+        }
+    }
+
+    pub fn save_root(&self, root: &H256) -> Result<(), SMTError> {
+        self.store
+            .insert_raw(self.root_col, &self.lock_hash, root.as_slice())
+            .map_err(|err| SMTError::Store(format!("insert error {:?}", err)))?;
+        Ok(())
+    }
+
+    pub fn get_root(&self) -> Result<Option<H256>, SMTError> {
+        match self.store.get(self.root_col, &self.lock_hash) {
+            Some(slice) => Ok(Some(H256::from(parse_vec_n(slice.to_vec())))),
+            None => Ok(None),
+        }
+    }
+
+    pub fn insert_leaves(&self, leaves: Vec<(H256, H256)>) -> Result<(), Error> {
+        let smt_leaves: Vec<SMTLeaf> = leaves
+            .into_iter()
+            .map(|leaf| SMTLeaf {
+                key:   leaf.0.as_slice().to_vec(),
+                value: leaf.1.as_slice().to_vec(),
+            })
+            .collect();
+        let json = serde_json::to_string(&smt_leaves)
```

Comment:
The json format adds unnecessary serialization costs, suggest to use a molecule format or customize simple serialization format.

```
vector SMTLeafVec <SMTLeaf>
struct SMTLeaf {
    key: Byte32,
    value: Byte32,
}
```

Reply:
OK